internal/provider/utils.go

```diff
@@ -1,9 +1,45 @@
 package provider
 
 import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"slices"
+
 	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
 )
 
 func IsValueProvided(value attr.Value) bool {
 	return !value.IsNull() && !value.IsUnknown()
 }
+
+func ValidateResponse(resp *http.Response, body []byte, err error, expected_statuses []int) diag.Diagnostics {
```

Comment:
I like this function overall, but we are unmarshalling the response twice. I feel like if we renamed it to `ParseResponse()` or something else more appropriate, it could return the marshalled response.

Reply:
Oh, I responded to @abikouo's comment above and then realized why I did not do this. It only gets unmarshalled here if the HTTP status code is not what we expect, so that we can provide the details from the body in the diagnostics error to explain why the unexpected status code was returned. In the inventory_resource `parseHTTPResponse` function it gets unmarshalled directly into the `inventoryApiModel` struct after we know we've received the expected response. So it's only ever getting unmarshalled once, either here _if_ there's an unexpected status code, or in the `parseHTTPResponse` function for each resource (where ultimately I'm hoping we will refactor to unmarshal into a known struct for all the resources).

Reply:
OK, that makes more sense. Do we know what the response body looks like when the status code isn't one of the expected status codes? Does it ever have anything useful in it in these cases?

Reply:
Yes, it has useful info at least some of the time. It can vary but provides details like "this resource already exists" or the reason for a bad request, or why there's an authentication error. I added this largely because it was helpful to me for debugging during development and I expect it will be similarly helpful for troubleshooting errors down the line.